### PR TITLE
 [805] Add sysml textual version of import unit tests resources.

### PR DESCRIPTION
**Pull request by ebausson-obeo**
Bug: #805
 Signed-off-by: Étienne Bausson <etienne.bausson@obeosoft.com>

# PLEASE READ ALL ITEMS AND CHECK ONLY RELEVANT CHECKBOXES BELOW

## Project management

- [x] Has the pull request been added to the relevant milestone?
- [x] Have the `priority:` and `pr:` labels been added to the pull request? (In case of doubt, start with the labels `priority: low` and `pr: to review later`)
- [x] Have the relevant issues been added to the pull request?
- [x] Have the relevant labels been added to the issues? (`area:`, `type:`)
- [x] Have the relevant issues been added to the same project milestone as the pull request?

## Changelog and release notes

- [X] Has the `CHANGELOG.adoc` + `doc/content/modules/user-manual/pages/release-notes/YYYY.MM.0.adoc` been updated to reference the relevant issues?
- [ ] Have the relevant API breaks been described in the `CHANGELOG.adoc` + `doc/content/modules/user-manual/pages/release-notes/YYYY.MM.0.adoc`?
- [ ] In case of a change with a visual impact, are there any screenshots in the `doc/content/modules/user-manual/pages/release-notes/YYYY.MM.0.adoc`?
- [ ] In case of a key change, has the change been added to `Key highlights` section in `doc/content/modules/user-manual/pages/release-notes/YYYY.MM.0.adoc`?
- [ ] Are the new / upgraded dependencies mentioned in the relevant section of the `CHANGELOG.adoc` + `doc/content/modules/user-manual/pages/release-notes/YYYY.MM.0.adoc`?

## Documentation

- [ ] Have you included an update of the [documentation](https://doc.mbse-syson.org) in your pull request? Please ask yourself if an update (installation manual, user manual, developer manual...) is needed and add one accordingly.

## Tests

- [ ] Is the code properly tested? Any pull request (fix, enhancement or new feature) should come with a test (or several). It could be unit tests, integration tests or cypress tests depending on the context. Only doc and releng pull request do not need for tests.

### CODE REVIEW

File: CHANGELOG.adoc [hunk 1]
```diff
@@ -61,6 +61,7 @@ This reduces the time required to build SysON, especially when using Spring Tool
 - https://github.com/eclipse-syson/syson/issues/785[#785] [interconnection-view] Add `New Binding Connector As Usage` tool on `PartUsage` in Interconnection View diagram.
 - https://github.com/eclipse-syson/syson/issues/789[#789] [interconnection-view] Add `New Flow Connection` tool on `PartUsage` in Interconnection View diagram.
 - https://github.com/eclipse-syson/syson/issues/790[#790] [interconnection-view] Add `New Interface` tool on `PartUsage` in Interconnection View diagram.
+- https://github.com/eclipse-syson/syson/issues/805[#805] [import] Improve non regression tests.
```
Comment (AxelRICHARD) [inline CHANGELOG.adoc:64]: This comment is too general. Could you be more specific please?

File: backend/application/syson-sysml-import/src/test/resources/ASTTransformerTest/convertNamespaceImportValueTest/model.sysml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+package package2 {
+    import package1::part1;
```
Comment (AxelRICHARD) [inline backend/application/syson-sysml-import/src/test/resources/ASTTransformerTest/convertNamespaceImportValueTest/model.sysml:2]: Could you please specify the visibility of the import?

File: backend/application/syson-sysml-import/src/test/resources/ASTTransformerTest/convertNamespaceImportTest/model.sysml [hunk 3]
```diff
@@ -0,0 +1,3 @@
+package Packets {
+    import ScalarValues;
```
Comment (AxelRICHARD) [inline backend/application/syson-sysml-import/src/test/resources/ASTTransformerTest/convertNamespaceImportTest/model.sysml:2]: Could you please specify the visibility of the import?

File: backend/application/syson-sysml-import/src/test/resources/ASTTransformerTest/convertAliasTest/alias.sysml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+package AliasImport {
+    package Definitions {
+        part def Vehicle;
+        alias Car for Vehicle;
+    }
+    package Usages {
+        import Definitions::Car;
```
Comment (AxelRICHARD) [inline backend/application/syson-sysml-import/src/test/resources/ASTTransformerTest/convertAliasTest/alias.sysml:7]: Could you please specify the visibility of the import?

**Review by AxelRICHARD — COMMENTED** [review]
Could you please change your commit title? 
Something like `Add sysml textual version of import unit tests resources`

File: backend/application/syson-sysml-import/src/test/resources/ASTTransformerTest/convertNamespaceImportValueTest/model.sysml (hunk 2 above)
Comment (ebausson-obeo) [reply backend/application/syson-sysml-import/src/test/resources/ASTTransformerTest/convertNamespaceImportValueTest/model.sysml:3]: Seems to be outside the scope of the test (for all 3 such request).
I agree that tests on the visibility of import are necessary, especially after the changes of 2.2 but shouldn't they be in their own test?

Comment (ebausson-obeo) [reply backend/application/syson-sysml-import/src/test/resources/ASTTransformerTest/convertNamespaceImportValueTest/model.sysml]: fait.

File: backend/application/syson-sysml-import/src/test/resources/ASTTransformerTest/convertNamespaceImportTest/model.sysml (hunk 3 above)
Comment (ebausson-obeo) [reply backend/application/syson-sysml-import/src/test/resources/ASTTransformerTest/convertNamespaceImportTest/model.sysml]: fait.

File: backend/application/syson-sysml-import/src/test/resources/ASTTransformerTest/convertAliasTest/alias.sysml (hunk 4 above)
Comment (ebausson-obeo) [reply backend/application/syson-sysml-import/src/test/resources/ASTTransformerTest/convertAliasTest/alias.sysml]: fait.

File: backend/application/syson-sysml-import/src/test/resources/ASTTransformerTest/convertAssignmentTest/model.ast.json [hunk 5]
```diff
@@ -0,0 +1,398 @@
+{
```
Comment (AxelRICHARD) [inline backend/application/syson-sysml-import/src/test/resources/ASTTransformerTest/convertAssignmentTest/model.ast.json:1]: This model doesn't seems to add its associated textual sysml file.
Is it intended?
(same remarks for all other similar cases)

File: backend/application/syson-sysml-import/src/test/resources/ASTTransformerTest/convertAliasTest/model.ast.json [hunk 6]
```diff
@@ -15,21 +15,99 @@
                   "$type": "OwningMembership",
```
Comment (AxelRICHARD) [inline backend/application/syson-sysml-import/src/test/resources/ASTTransformerTest/convertAliasTest/model.ast.json:15]: Why this file is named model.ast.json ? Shouldn't it be alias.ast.json instead?
(same remarks for all other similar cases)

File: CHANGELOG.adoc [hunk 7]
```diff
@@ -80,6 +80,7 @@ This reduces the time required to build SysON, especially when using Spring Tool
 - https://github.com/eclipse-syson/syson/issues/790[#790] [interconnection-view] Add `New Interface` tool on `PartUsage` in Interconnection View diagram.
 - https://github.com/eclipse-syson/syson/issues/799[#799] [export] Handle `UseCaseUsage` and `IncludeUseCaseUsage` elements.
 - https://github.com/eclipse-syson/syson/issues/803[#803] [libraries] All standard libraries have been updated to comply with the SysML Beta 2.2 specification.
+- https://github.com/eclipse-syson/syson/issues/805[#805] [import] Improve non regression tests. Added source sysml file and unaltered ast.json result.
```
Comment (AxelRICHARD) [inline CHANGELOG.adoc:83]: Could you please add a new line between the two sentences?
We let only one sentence per line.

File: backend/application/syson-sysml-import/src/test/resources/ASTTransformerTest/convertAssignmentTest/model.ast.json (hunk 5 above)
Comment (ebausson-obeo) [reply backend/application/syson-sysml-import/src/test/resources/ASTTransformerTest/convertAssignmentTest/model.ast.json:1]: artefact of my creation script. removed.

File: backend/application/syson-sysml-import/src/test/resources/ASTTransformerTest/convertAliasTest/model.ast.json (hunk 6 above)
Comment (ebausson-obeo) [reply backend/application/syson-sysml-import/src/test/resources/ASTTransformerTest/convertAliasTest/model.ast.json]: done

File: CHANGELOG.adoc (hunk 7 above)
Comment (ebausson-obeo) [reply CHANGELOG.adoc]: done.

**Pull request merged**